include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
OK.   I guess, I would feel more confortable if the test wasn't for `_HAS_STD_BYTE :-)
@Casey: _MSC_VER_FULL is too finegrained